apps/wallet/src/utils/hooks/navigation.ts

```diff
@@ -21,10 +21,21 @@ export const useNavigationHydrate = () => {
 			/* eslint-disable */
 			const findDashboard = (i: any) => i.name === 'Dashboard';
 			const dashboard = state?.routes.find(findDashboard)?.state;
-			const childRoute = dashboard?.routes[dashboard?.index as never];
-			const widget = childRoute?.params as any;
-			const widgetId = widget?.params?.id;
-			/* eslint-enable */
+			const childRoute = dashboard?.routes[dashboard?.index as never] as {
+				key: string;
+				name: string;
+				params: {
+					screen: string;
+					params: {
+						id: string;
+					};
+				};
+			};
+
+			const widgetId =
+				childRoute?.params.screen === 'Collectible' || 'Collection'
+					? ''
+					: childRoute.params?.params?.id;
 
 			if (childRoute?.name === 'Explore' && widgetId !== lastWidgetId) {
```

Comment:
I find that there is a simpler way, just keep `widget` variable, and add an additional condition here `widget.screen === 'Widget'` because what we want is ignore all routes except widget route.

Reply:
You mean this?

```ts
const widgetId = childRoute?.params?.screen === 'Widget'
	? childRoute.params?.params.id
	: '';
```